sdks/java/io/hcatalog/src/test/java/org/apache/beam/sdk/io/hcatalog/HCatalogIOTest.java

```diff
@@ -0,0 +1,342 @@
+ */
+
+package org.apache.beam.sdk.io.hcatalog;
+
+import static org.apache.beam.sdk.io.hcatalog.HCatalogIOTestUtils.TEST_TABLE_NAME;
+import static org.apache.beam.sdk.io.hcatalog.HCatalogIOTestUtils.commitRecords;
+import static org.apache.beam.sdk.io.hcatalog.HCatalogIOTestUtils.getReaderContext;
+import static org.apache.beam.sdk.io.hcatalog.HCatalogIOTestUtils.getWriterContext;
+import static org.apache.beam.sdk.io.hcatalog.HCatalogIOTestUtils.toHCatRecord;
+import static org.apache.beam.sdk.io.hcatalog.HCatalogIOTestUtils.writeRecords;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.isA;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.beam.sdk.io.hcatalog.HCatalogIO.BoundedHCatalogSource;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.SourceTestUtils;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.util.UserCodeException;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.hadoop.hive.ql.CommandNeedRetryException;
+import org.apache.hive.hcatalog.common.HCatException;
+import org.apache.hive.hcatalog.data.DefaultHCatRecord;
+import org.apache.hive.hcatalog.data.transfer.ReaderContext;
+import org.apache.hive.hcatalog.data.transfer.WriterContext;
+import org.apache.hive.hcatalog.mapreduce.HCatBaseTest;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+
+/**
+ *Test for HCatalogIO.
+ */
+public class HCatalogIOTest extends HCatBaseTest implements Serializable {
+
+  @Rule
+  public final transient TestPipeline pipeline = TestPipeline.create();
+
+  @Rule
+  public transient ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  /**
+   * Perform end-to-end test of Read operation.
+   */
+  public void testReadE2ESuccess() throws CommandNeedRetryException,
+  IOException, ClassNotFoundException {
+
+    Map<String, String> map = insertTestData();
+
+    PCollection<String> output = pipeline.apply(HCatalogIO.read()
+        .withConfigProperties(map)
+        .withTable(HCatalogIOTestUtils.TEST_TABLE_NAME))
+        .apply(ParDo.<DefaultHCatRecord, String>of(new DoFn<DefaultHCatRecord, String>() {
+          @ProcessElement
+          public void processElement(ProcessContext c) {
+            c.output(c.element().get(0).toString());
+          }
+        }));
+    PAssert.thatSingleton(output.apply("Count All", Count.<String>globally()))
+    .isEqualTo(100L);
```

Comment:
please do a better check than just count.  PAssert.that(collection).containsInAnyOrder() will do what you want.